### PR TITLE
Fix line info for import 

### PR DESCRIPTION
**Pull request by ire4ever1190**
Refs #24158

Fixes the line info of the module symbol (cases like `import as` and grouped imports had wrong line info). Since that symbol's line info is now used for the warnings, there isn't a separate line info stored for `unusedImports`

Examples of fixed cases
```nim
import strutils as test #[
                ^ before
                   ^ after ]#

# This case was fixed by #24158, but only for unused imports
import std/[strutils, strutils] #[
        ^ before
                      ^ after ]#

from strutils import split #[
^ before 
     ^ after ]#
```

### CODE REVIEW

**Comment by ire4ever1190** [comment]
~~Errors related, will fix up the other tests~~
Done

File: compiler/semdata.nim [hunk 1]
```diff
@@ -164,7 +164,7 @@ type
     isAmbiguous*: bool # little hack
     features*: set[Feature]
     inTypeContext*, inConceptDecl*: int
-    unusedImports*: seq[(PSym, TLineInfo)]
+    unusedImports*: seq[PSym]
```
Comment (Araq) [inline compiler/semdata.nim]: If you use the lineinfo from the PSym you rely on the fact that module PSym are duplicated. But I'm not sure we want to commit to this design choice.

File: compiler/importer.nim [hunk 2]
```diff
@@ -335,7 +336,7 @@ proc impMod(c: PContext; it: PNode; importStmtResult: PNode) =
   let m = myImportModule(c, it, importStmtResult)
   if m != nil:
     # ``addDecl`` needs to be done before ``importAllSymbols``!
-    addDecl(c, m, it.info) # add symbol to symbol table of module
+    addDecl(c, m) # add symbol to symbol table of module
```
Comment (Araq) [inline compiler/importer.nim:339]: Seems better to use `addDecl(c, m, m.info)` here and elsewhere and keep the line information explicit in the data model.

File: compiler/semdata.nim (hunk 1 above)
Comment (ire4ever1190) [reply compiler/semdata.nim]: Done

File: compiler/importer.nim (hunk 2 above)
Comment (ire4ever1190) [reply compiler/importer.nim:339]: The `addDecl` overload here gets the line info from `m`.
Or do you more mean get the line info here so its more clear that its coming from `m`?

**Pull request merged**